src/control/cmd/daos_agent/procmon.go

```diff
@@ -342,5 +345,53 @@ func (p *procMon) handleRequests(ctx context.Context) {
 // startMonitoring is the main driver which starts the process monitor. The
 // passed in context is used to terminate all monitoring in the event of shutdown.
 func (p *procMon) startMonitoring(ctx context.Context) {
+	if p.cleanOnStart {
+		p.cleanupServerHandles(ctx)
+	}
 	go p.handleRequests(ctx)
 }
+
+// cleanupServerHandles can be run to revoke all pool handles associated with a given machine/host.
+// DAOS server will be instructed to cleanup handles associated with the source machine/host.
+func (p *procMon) cleanupServerHandles(ctx context.Context) {
+	hostName, err := os.Hostname()
+	if err != nil {
+		p.log.Errorf("hostname lookup: %s, cannot cleanup handles", err)
+		return
+	}
+	// Strip the domain off of the Hostname
+	machineName := strings.Split(hostName, ".")[0]
+	if machineName == "" {
+		p.log.Errorf("empty machine name is invalid, cannot cleanup handles", err)
+		return
+	}
+
+	req := &control.SystemCleanupReq{Machine: machineName}
+	req.SetSystem(p.systemName)
+
+	msg := fmt.Sprintf("machine %q", machineName)
+	resp, err := control.SystemCleanup(ctx, p.ctlInvoker, req)
+	if err != nil {
+		p.log.Errorf("%s: failed to run system cleanup: %s", msg, err)
+		return
+	}
+	err = resp.Errors()
+	if err != nil {
+		p.log.Errorf("%s: system cleanup ran with error: %s", msg, err)
+		return
+	}
+
+	msg = fmt.Sprintf("Running system cleanup on this %s", msg)
+	if len(resp.Results) == 0 {
+		p.log.Infof("%s: no pool handles revoked", msg)
```

Comment:
Probably better as debug, no? I don't think this is relevant to normal operations.

Reply:
done